receiver/awscontainerinsightreceiver/internal/stores/kubeletutil/kubeletclient.go

```diff
@@ -0,0 +1,104 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeletutil
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/tls"
+)
+
+type KubeClient struct {
+	Port            string
+	BearerToken     string
+	KubeIP          string
+	responseTimeout time.Duration
+	roundTripper    http.RoundTripper
+	tls.ClientConfig
+}
+
+var ErrKubeClientAccessFailure = errors.New("KubeClinet Access Failure")
+
+func (k *KubeClient) ListPods() ([]corev1.Pod, error) {
+	var result []corev1.Pod
+	var req *http.Request
+	url := fmt.Sprintf("https://%s:%s/pods", k.KubeIP, k.Port)
+
+	req, _ = http.NewRequest("GET", url, nil)
+	var resp *http.Response
+
+	k.InsecureSkipVerify = true
+	tlsCfg, err := k.ClientConfig.TLSConfig()
```

Comment:
This is not OK.

Reply:
I am now using the kubelet client from `kubeletstatsreceiver` and configure it to use pod service account for authentication.